apps/condo/domains/ticket/tasks/exportTicketsTask.js

```diff
@@ -0,0 +1,207 @@
+    }
+}
+
+const buildExportFile = async ({ rows, task, idOfFirstTicketForAccessRights }) => {
+    const { where, timeZone, locale } = task
+    const createdAtGte = get(findAllByKey(where, 'createdAt_gte'), 0)
+    const createdAtLte = get(findAllByKey(where, 'createdAt_lte'), 0)
+
+    const formatDate = (date) => dayjs(date).tz(timeZone).format(HEADER_DATE_FORMAT)
+
+    let headerMessage = i18n('excelExport.header.tickets.forAllTime', { locale })
+    if (createdAtGte && createdAtLte) {
+        headerMessage = `${i18n('excelExport.header.tickets.forPeriod', { locale })} ${formatDate(createdAtGte)} — ${formatDate(createdAtLte)}`
+    }
+
+    const reviewValuesTranslations = buildReviewValuesTranslationsFrom(locale)
+
+    const { stream } = await _buildExportFile({
+        templatePath: './domains/ticket/templates/TicketsExportTemplate.xlsx',
+        replaces: {
+            header: headerMessage,
+            tickets: rows,
+            i18n: {
+                ...getHeadersTranslations(EXPORT_TYPE_TICKETS, locale),
+                sheetName: i18n('excelExport.sheetNames.tickets', { locale }),
+
+                // These statuses are needed to set cell color using conditional formatting by status name.
+                statusNames: ticketStatusesTranslations(locale),
+
+                yes: i18n('Yes', { locale }),
+                reviewValues: reviewValuesTranslations,
+            },
+        },
+    })
+    return {
+        stream,
+        filename: `tickets_${dayjs().format('DD_MM')}.xlsx`,
+        mimetype: EXCEL_FILE_META.mimetype,
+        encoding: EXCEL_FILE_META.encoding,
+        meta: {
+            listkey: 'Ticket',
+            // Id of first record will be used by `OBSFilesMiddleware` to determine permission to access exported file
+            // NOTE: Permissions check on access to exported file will be replaced to checking access on `ExportTicketsTask`
+            id: idOfFirstTicketForAccessRights,
+        },
+    }
+}
+
+const exportTickets = async (taskId) => {
+    if (!taskId) throw new Error('taskId is undefined')
+    const { keystone: context } = await getSchemaCtx('TicketExportTask')
+
+    const task = await TicketExportTask.getOne(context, { id: taskId })
+
+    const statuses = await TicketStatus.getAll(context, {})
+    const indexedStatuses = Object.fromEntries(statuses.map(status => ([status.type, status.name])))
+
+    let idOfFirstTicketForAccessRights
+
+    let classifierRules
+
+    await exportRecords({
+        context,
+        loadRecordsBatch: async (offset, limit) => {
+            const { where, sortBy } = task
+            const tickets = await loadTicketsBatchForExcelExport({ where, sortBy, offset, limit })
+            classifierRules = await loadClassifiersForExcelExport({ rulesIds: map(tickets, 'classifierRule') })
+
+            if (!idOfFirstTicketForAccessRights) {
+                idOfFirstTicketForAccessRights = get(tickets[0], 'id')
```

Comment:
It is safer to get it like suggest below
```suggestion
                idOfFirstTicketForAccessRights = get(tickets, '0.id')
```